frontend/src/metabase/query_builder/components/expressions/ExpressionEditorTextfield/ExpressionEditorTextfield.tsx

```diff
@@ -462,9 +462,9 @@ class ExpressionEditorTextfield extends React.Component<
     if (this.props.onBlankChange) {
       this.props.onBlankChange(source.length === 0);
     }
-  }
+  };
 
-  handleCursorChange(selection: Ace.Selection) {
+  handleCursorChange = _.debounce((selection: Ace.Selection) => {
```

Comment:
probably the most important, change, based on my investigations, `ExpressionEditorTextfield` component is rendered 3 times per keypress, most likely because of separate updates of the state. I spent some time trying to fix the reason, but didn't find how to fix it. 

Anyways, having a small delay between typing and asking for columns from MLv2 methods reduces number of requests from 3 to 1